solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -725,7 +730,7 @@ protected void search(List<LeafReaderContext> leaves, Weight weight, Collector c
       // So we need to make a new IndexSearcher instead of using "this".
       new IndexSearcher(reader) { // cheap, actually!
         void searchWithTimeout() throws IOException {
-          setTimeout(queryTimeout.makeLocalImpl());
+          setTimeout(requestInfo.getLimits());
           super.search(leaves, weight, collector); // FYI protected access
           if (timedOut()) {
             throw new TimeAllowedExceededFromScorerException("timeAllowed exceeded");
```

Comment:
Hmm, perhaps we should change this message to "allowed limit exceeded". Which suggests that we should be able to tell user what limit has been exceeded if multiple limits have been specified... We could add a utility method to `QueryLimits` to format a string telling what kind of limit has been tripped, based on the current values of `shouldExit` of each impl.